packages/transport/src/modules/relationships/RelationshipsController.ts

```diff
@@ -423,218 +335,141 @@ export class RelationshipsController extends TransportController {
-            throw CoreErrors.relationships.wrongChangeStatus(queriedChange.status);
-        }
-
-        let encryptedContent;
-        if (content) {
-            encryptedContent =
-                targetStatus === RelationshipChangeStatus.Revoked
-                    ? await this.encryptRevokeContent(relationship, content)
-                    : await this.encryptAcceptRejectContent(relationship, content);
-        }
-
-        let backboneResponse: BackboneGetRelationshipsResponse;
-        switch (targetStatus) {
-            case RelationshipChangeStatus.Accepted:
-                backboneResponse = (await this.client.acceptRelationshipChange(relationship.id.toString(), change.id.toString(), encryptedContent)).value;
-                break;
-
-            case RelationshipChangeStatus.Rejected:
-                backboneResponse = (await this.client.rejectRelationshipChange(relationship.id.toString(), change.id.toString(), encryptedContent)).value;
-                break;
-
-            case RelationshipChangeStatus.Revoked:
-                backboneResponse = (await this.client.revokeRelationshipChange(relationship.id.toString(), change.id.toString(), encryptedContent)).value;
-                break;
-
-            default:
-                throw new TransportError("target change status not supported");
+            const newRelationship = await this.createNewRelationshipByIncomingCreation(relationshipId);
+            if (newRelationship.status === RelationshipStatus.Pending) {
+                return newRelationship;
+            }
+            // this path is for a revocation that is processed before its corresponding creation
+            relationshipDoc = await this.relationships.read(relationshipId);
         }
-        const backboneChange = backboneResponse.changes[backboneResponse.changes.length - 1];
 
-        queriedChange.response = RelationshipChangeResponse.fromBackbone(backboneResponse.changes[backboneResponse.changes.length - 1].response!, content);
-        queriedChange.status = backboneChange.status;
-        relationship.status = backboneResponse.status;
-
-        await this.relationships.update(relationshipDoc, relationship);
-
-        this.eventBus.publish(new RelationshipChangedEvent(this.parent.identity.address.toString(), relationship));
-
-        return relationship;
+        return await this.updatePendingRelationshipWithPeerResponse(relationshipDoc);
     }
+    private async encryptAcceptanceContent(relationship: Relationship) {
+        const publicResponseCrypto = await this.secrets.getPublicResponse(relationship.relationshipSecretId);
 
-    private async encryptRevokeContent(relationship: Relationship, content: ICoreSerializable) {
-        const responseContent = RelationshipCreationChangeResponseContentWrapper.from({
+        const responseContent = RelationshipCreationResponseContentWrapper.from({
             relationshipId: relationship.id,
-            content: content
+            content: ""
```

Comment:
you removed the content, why is that?

Reply:
to stick with the original plan of having no acceptanceContent

Reply:
why don't you remove it everywhere then?

Reply:
I only wanted to keep as much as necessary for the secret transfer - did I miss something?

Reply:
Maybe I miss something. We should talk about this.

Reply:
discussed - we do not send an custom acceptance content currently, but we need the acceptance content for key exchange.